### ISSUE TITLE
Speed considerations

### ISSUE THREAD

**Issue by LinearParadox** [aet21/ELVAR#2]
A single run of eva.o takes around 15-20 mins. To run it 100 times (as suggested in the manual) would obviously take a pretty long time. I was wondering if there are any parallelization or speedup options available?

**Comment by aet21** [comment]
Yes, you are right, if you have lots of cells, one run can take this long, but nowadays we do normally have access to servers with 100 nodes or so, and so 100 runs are achievable within 30 mins. To run in parallel mode should be quite easy with the parallel R-library, but I agree for those unfamiliar with this, we should make the parallel option available within the package. We will work on it and update package fairly soon. If you can't wait,
a quick fix would be to refine Eva_partitions function (call this new function myEva_partitions) so that the 1st argument is just a dummy integer index, then you could just do
library(parallel);
nR <- 100; ## number of runs desired
idx.l <- as.list(1:nR);
mcl.o <- mclapply(idx.l,myEva_partitions,all other arguments here,mc.cores=100);
each entry of mcl.lo contain the output of one Eva run

kr
A

*************************************************************
Professor Andrew E Teschendorff
Computational Systems Epigenomics Lab
CAS Key Lab of Computational Biology
Shanghai Institute for Nutrition and Health
Chinese Academy of Sciences
320 Yueyang Road
Shanghai 200031
Tel: +86 183-1704-7442
email: ***@***.***
web: https://aeteschendorff-lab.github.io
**************************************************************
________________________________
From: ***@***.*** ***@***.***> on behalf of Alexei Martsinkovskiy ***@***.***>
Sent: Saturday, July 15, 2023 7:11 AM
To: aet21/ELVAR ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [aet21/ELVAR] Speed considerations (Issue #2)


A single run of eva.o takes around 15-20 mins. To run it 100 times (as suggested in the manual) would obviously take a pretty long time. I was wondering if there are any parallelization or speedup options available?
